### PR TITLE
SOCKSQL: defer sock start until we have to send something to master

### PR DESCRIPTION
**Pull request by ldennis**
SOCKSQL: defer sock start until we have to send something to master

### CODE REVIEW

**Comment by nirbhayc** [comment]
+399.89%  whoa!!

**Comment by markhannum** [comment]
OH MY GOD HM!!!!

File: db/osqlsqlthr.c [hunk 1]
```diff
@@ -774,6 +801,10 @@ int osql_sock_start(struct sqlclntstate *clnt, int type, int keep_rqid)
 
     if (!keep_rqid && rc != 0) {
         osql_unregister_sqlthr(clnt);
+    } else {
```
Comment (riverszhang89) [inline db/osqlsqlthr.c]:
```c
if (rc == 0) {
    if (clnt->client_understands_query_stats)
            osql_query_dbglog(thd, clnt->queryid);
    osql->sock_started = 1;
} else if (!keep_rqid)
    osql_unregister_sqlthr(clnt);
```

**Comment by ldennis** [comment]
in-comdb2db defer_sock.R20190214.1

**Pull request merged**